### ISSUE TITLE
How to build dist?

### ISSUE THREAD

**Issue by drnic** [starkandwayne/www-static#1]
@mlbcreative I guessed the following incorrectly. What are the steps for building?

```
$ foundation build

> foundation-zurb-template@1.0.0 build /Users/drnic/Projects/starkandwayne/www-static
> gulp build --production

[16:22:02] Using gulpfile ~/Projects/starkandwayne/www-static/gulpfile.js
[16:22:02] Task 'build' is not in your gulpfile
[16:22:02] Please check the documentation for proper gulpfile formatting
Error: foundation-zurb-template@1.0.0 build: `gulp build --production`
Exit status 1
    at EventEmitter.<anonymous> (/usr/local/lib/node_modules/foundation-cli/node_modules/npm/lib/utils/lifecycle.js:217:16)
    at emitTwo (events.js:87:13)
    at EventEmitter.emit (events.js:172:7)
    at ChildProcess.<anonymous> (/usr/local/lib/node_modules/foundation-cli/node_modules/npm/lib/utils/spawn.js:24:14)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:827:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
```

**Comment by drnic** [comment]
Ooh, just found `readme_mlb.md` and am reading, sorry.

**Comment by drnic** [comment]
Ok, commands that worked for me:

```
npm install foundation-cli --global
npm rebuild node-sass
npm start
```

But, this runs it - rather than completes and exits. How do I just "build the site" (as we will do in a CI pipeline)?

Also, it deleted `dist/Staticfile`. We need to keep this file (or recreate it I guess) for CI.

Also, it created some new files that weren't there before; so perhaps I'm doing something wrong?

New files:

```
    dist/assets/
    dist/index copy.html
    dist/styleguide.html
    node_modules/gulp-sass/node_modules/node-sass/vendor/darwin-x64-46/
```

**Comment by mlbcreative** [comment]
Hey there,
Make sure you're inside the WWW-static folder and run foundation watch

Charles Rosenberger
Sr. Digital Director
MLB Creative
http://mlbcreative.com

**Comment by drnic** [comment]
@mlbcreative this is like `npm start` in that it starts a long running process/never terminates. How do I build the site without starting a process? (like you would in a CI/jenkins system prior to running `cf push`)?

**Comment by drnic** [comment]
Also, `foundation watch` also created new files that weren't in the git repo; plus deleted `dist/Staticfile` as commented above.

I guess the new files in `dist` are supposed to be there in the git repo? Or should I put `dist` in `.gitignore`?

**Comment by drnic** [comment]
I think I want `foundation build` to work. Can that be supported?

**Comment by mlbcreative** [comment]
Foundation watch starts a process that compiles all the files in the src folder, deletes the dist folder, creates a new dist folder and puts all the compiled Files into the dist folder. It then watched for any changes to the files. The contents of dist are what should be deployed to the web server. 

I usually run fountain watch , build the site and the. Ctrl + C out of the process. Hope that helps. 

Charles Rosenberger
Sr. Digital Director
MLB Creative
http://mlbcreative.com

**Comment by mlbcreative** [comment]
In all honesty I'm not sure. We always run the watch command. Sorry I'm not more help in that front. 

Charles Rosenberger
Sr. Digital Director
MLB Creative
http://mlbcreative.com

**Comment by mlbcreative** [comment]
Hey Dr. Nic,

I was dictating through my phone so I apologize for the run on sentences of my previous emails. If you'd like I can run an initial build. Here's our dev process - if it helps

we don't run any npm commands at all, just the foundation specific commands in the foundation cli (watch, update, etc)
1. I cd into the root directory
2. run foundation watch
3. The process starts and also starts a browsersync session. It should pop open a browser window for you with this url http://localhost:8000
4. Edits are made to the files in the src directory
5. The proccess looks for changes and recompiles and refreshes the browser automatically.
6. I ctrl+c in the terminal to stop the process.
   Hope that helps

The downside to this process is that it deletes the dist folder and then recreates it on the fly. If you look at the gulpfile I believe you can comment out the command that deletes the dist folder.

**Comment by drnic** [comment]
@mlbcreative I can appreciate that interactive `foundation watch` works just perfectly for designers, not a problem. Our future deployment flow will be via CI so need a non-interactive command to run.

If your team isn't sure how to add the missing `build` bits (as per the error), I'll see if someone from @starkandwayne/engineering  does.

**Comment by mlbcreative** [comment]
Totally understand. I can look into it but I won't stop you if you need to loop your team in. 😁

Charles Rosenberger
Sr. Digital Director
MLB Creative
http://mlbcreative.com

**Comment by krutten** [comment]
created https://github.com/starkandwayne/www-static/pull/2 as a possible solution

```
[ruby-2.2.3@bosh] (add-build) Homer:www-static krutten$ foundation build

> foundation-zurb-template@1.0.0 build /Users/krutten/StarkAndWayne/projects/node/www-static
> gulp build --production

[17:46:39] Using gulpfile ~/StarkAndWayne/projects/node/www-static/gulpfile.js
[17:46:39] Starting 'build'...
[17:46:39] Starting 'clean'...
[17:46:39] Finished 'clean' after 15 ms
[17:46:39] Starting 'pages'...
[17:46:39] Finished 'pages' after 177 ms
[17:46:39] Starting 'sass'...
[17:46:40] Starting 'javascript'...
[17:46:40] Starting 'images'...
[17:46:41] Starting 'copy'...
[17:46:41] Finished 'copy' after 430 μs
[17:46:49] Finished 'javascript' after 9.34 s
InvalidValueError: initMap is not a function

  https://maps.googleapis.com/maps/api/js?key=ZZZZZZZZZ&callback=initMap:94
  https://maps.googleapis.com/maps/api/js?key= ZZZZZZZZZ&callback=initMap:131
  https://maps.googleapis.com/maps/api/js?key= ZZZZZZZZZ&callback=initMap:21
  https://maps.googleapis.com/maps/api/js?key= ZZZZZZZZZ&callback=initMap:131
  https://maps.googleapis.com/maps/api/js?key= ZZZZZZZZZ&callback=initMap:131
[17:46:50] gulp-imagemin: Minified 72 images (saved 41.61 kB - 1.4%)
[17:46:50] Finished 'images' after 9.66 s
[17:46:51] Finished 'sass' after 11 s
[17:46:51] Starting 'styleguide'...
[17:46:51] Finished 'styleguide' after 35 ms
[17:46:51] Finished 'build' after 12 s
[]
[ruby-2.2.3@bosh] (add-build) Homer:www-static krutten$
```

**Comment by drnic** [comment]
Closed by #2 thanks @krutten
